packages/core/src/components/axes/tool-bar.ts

```diff
@@ -0,0 +1,325 @@
+			// add zoom in/out button only if zoom bar is enabled
+			if (this.isZoomBarEnabled) {
+				buttonList.push(this.getZoomInButtonConfig());
+				buttonList.push(this.getZoomOutButtonConfig());
+			}
+			buttonList.push(this.getOverFlowMenuButtonConfig());
+
+			// render buttons sequentially
+			let buttonXPosition = 0;
+			buttonList.forEach((button) => {
+				// zoom in icon and event
+				const buttonContainer = DOMUtils.appendOrSelect(
+					container,
+					`svg.${button.id}`
+				);
+				// add icon back ground rect
+				let buttonSVG = `<rect class="icon-${button.iconId}" x="${buttonXPosition}px" y="0px"
+					width="${iconSize}px" height="${iconSize}px"/>`;
+				buttonSVG += button.iconSVG(buttonXPosition);
+				buttonContainer.html(buttonSVG);
+				buttonContainer.on("click", button.clickFunction);
+				buttonXPosition += iconSize;
+			});
+
+			this.overflowMenuIcon = DOMUtils.appendOrSelect(
+				this.getContainerSVG(),
+				"rect.icon-overflowRect"
+			);
+
+			const hasOpenedOverflowMenuOptions =
+				this.overflowMenuOptions
+					.selectAll("ul.bx--overflow-menu-options--open")
+					.size() > 0;
+
+			document.body.addEventListener("click", function () {
+				// always clear menu icon hover state
+				self.setOverflowMenuIconHover(false);
+				self.services.events.dispatchEvent(
+					Events.Toolbar.HIDE_MENU_ITEMS
+				);
+			});
+
+			if (hasOpenedOverflowMenuOptions) {
+				this.setOverflowMenuIconHover(true);
+				this.overflowMenuOptions.html(this.getOverflowMenuHTML());
+			} else {
+				this.setOverflowMenuIconHover(false);
+				this.overflowMenuOptions.html(null);
+			}
+		}
+	}
+
+	getOverflowMenuHTML() {
+		// supports only reset zoom for now
+		// if getResetZoomMenuItem() return nothing
+		// don't render whole overflow menu
+		if (!this.getResetZoomMenuItem()) {
+			return "";
+		}
+
+		let defaultHTML;
+		defaultHTML = `<div data-floating-menu-container="true"
+			data-floating-menu-direction="bottom" role="main">
+			<ul class="bx--overflow-menu-options bx--overflow-menu--flip bx--overflow-menu-options--open"
+				tabindex="-1" role="menu" aria-label="Menu" data-floating-menu-direction="bottom"
+				style="left:${this.overflowMenuX}px; top:${this.overflowMenuY}px;">`;
+		defaultHTML += this.getResetZoomMenuItem();
+		defaultHTML += `</ul></div>`;
+
+		return defaultHTML;
+	}
+
+	getResetZoomMenuItem() {
+		const resetZoomOption = Tools.getProperty(
+			this.model.getOptions(),
+			"toolBar",
+			"toolBarMenuItems",
+			"resetZoom"
+		);
+		if (!resetZoomOption.enabled || !this.isZoomBarEnabled) {
+			return "";
+		} else {
+			return `<li
+						class="bx--overflow-menu-options__option">
+						<button class="bx--overflow-menu-options__btn" role="menuitem"  title="Reset"
+							data-floating-menu-primary-focus
+							id="${this.resetZoomMenuItemId}">
+							<div class="bx--overflow-menu-options__option-content">
+								${resetZoomOption.text}
+							</div>
+						</button>
+					</li>`;
+		}
+	}
```

Comment:
un-reusable code here^^. it only works with the zoom reset button, not framework to be able to support more buttons

Reply:
The new toolbar button / overflow menu item framework is updated in https://github.com/carbon-design-system/carbon-charts/pull/717/commits/d979d096cdc93aa14af8260fa60958ab89489955 and https://github.com/carbon-design-system/carbon-charts/pull/717/commits/c3d15a616d43e18bdd67ac0d5d7af1d747713222.